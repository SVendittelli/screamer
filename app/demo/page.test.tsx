import { render, screen } from "@testing-library/react";
import Demo from "./page";

describe("Demo", () => {
  it("renders a page with title 'Work in Progress'", () => {
    render(<Demo />);
    const title = screen.getByText(/Work in Progress/);
    expect(title).toBeInTheDocument();
  });

  it("shows the JSON data in a collapsed details tag", () => {
    render(<Demo />);
    const jsonData = screen.getByTestId("json-data");
    expect(jsonData).toBeInTheDocument();
    expect(jsonData).not.toBeVisible();

    const summary = screen.getByText(/JSON Data/);
    expect(summary).toBeInTheDocument();
    expect(summary).toBeVisible();
    summary.click();

    expect(jsonData).toBeVisible();
    expect(jsonData).toMatchSnapshot();
  });
});
