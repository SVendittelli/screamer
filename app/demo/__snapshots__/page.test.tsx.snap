// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Demo shows the JSON data in a collapsed details tag 1`] = `
<pre
  data-testid="json-data"
>
  [
  {
    "title": "Alien",
    "year": 1979,
    "poster": "https://m.media-amazon.com/images/M/MV5BOGQzZTBjMjQtOTVmMS00NGE5LWEyYmMtOGQ1ZGZjNmRkYjFhXkEyXkFqcGdeQXVyMjUzOTY1NTc@._V1_SX300.jpg",
    "plot": "The crew of a commercial spacecraft encounters a deadly lifeform after investigating an unknown transmission.",
    "rated": "R",
    "runtime": "117 min",
    "genre": "Horror, Sci-Fi",
    "director": "Ridley Scott",
    "writer": "Dan O'Bannon, Ronald Shusett",
    "actors": "Sigourney Weaver, Tom Skerritt, John Hurt",
    "language": "English",
    "awards": "Won 1 Oscar. 18 wins & 22 nominations total",
    "ratings": [
      {
        "source": "Internet Movie Database",
        "value": "8.5/10",
        "score": 8.5
      },
      {
        "source": "Rotten Tomatoes",
        "value": "98%",
        "score": 98
      },
      {
        "source": "Metacritic",
        "value": "89/100",
        "score": 89
      }
    ],
    "imdbVotes": "926,025",
    "imdbID": "tt0078748",
    "boxOffice": "$81,900,459"
  },
  {
    "title": "Let the Right One In",
    "year": 2008,
    "poster": "https://m.media-amazon.com/images/M/MV5BMjFjYmQ5OWItYTFiNy00Nzc5LTkyZGQtYmE0MDRhMjUwY2I5XkEyXkFqcGdeQXVyNTIzOTk5ODM@._V1_SX300.jpg",
    "plot": "Oskar, an overlooked and bullied boy, finds love and revenge through Eli, a beautiful but peculiar girl.",
    "rated": "R",
    "runtime": "114 min",
    "genre": "Drama, Fantasy, Horror",
    "director": "Tomas Alfredson",
    "writer": "John Ajvide Lindqvist",
    "actors": "KÃ¥re Hedebrant, Lina Leandersson, Per Ragnar",
    "language": "Swedish, Spanish",
    "awards": "Nominated for 1 BAFTA Award76 wins & 58 nominations total",
    "ratings": [
      {
        "source": "Internet Movie Database",
        "value": "7.8/10",
        "score": 7.8
      },
      {
        "source": "Rotten Tomatoes",
        "value": "98%",
        "score": 98
      },
      {
        "source": "Metacritic",
        "value": "82/100",
        "score": 82
      }
    ],
    "imdbVotes": "223,610",
    "imdbID": "tt1139797",
    "boxOffice": "$2,122,065"
  },
  {
    "title": "Aliens",
    "year": 1986,
    "poster": "https://m.media-amazon.com/images/M/MV5BOGJkY2EyOWYtYWRmNy00ZTEzLTllMDAtYzYzYjA0ZjFhZWJjXkEyXkFqcGdeQXVyMTUzMDUzNTI3._V1_SX300.jpg",
    "plot": "Decades after surviving the Nostromo incident, Ellen Ripley is sent out to re-establish contact with a terraforming colony but finds herself battling the Alien Queen and her offspring.",
    "rated": "R",
    "runtime": "137 min",
    "genre": "Action, Adventure, Sci-Fi",
    "director": "James Cameron",
    "writer": "James Cameron, David Giler, Walter Hill",
    "actors": "Sigourney Weaver, Michael Biehn, Carrie Henn",
    "language": "English, Spanish",
    "awards": "Won 2 Oscars. 20 wins & 25 nominations total",
    "ratings": [
      {
        "source": "Internet Movie Database",
        "value": "8.4/10",
        "score": 8.4
      },
      {
        "source": "Rotten Tomatoes",
        "value": "98%",
        "score": 98
      },
      {
        "source": "Metacritic",
        "value": "84/100",
        "score": 84
      }
    ],
    "imdbVotes": "747,025",
    "imdbID": "tt0090605",
    "boxOffice": "$85,160,248"
  },
  {
    "title": "Jaws",
    "year": 1975,
    "poster": "https://m.media-amazon.com/images/M/MV5BMmVmODY1MzEtYTMwZC00MzNhLWFkNDMtZjAwM2EwODUxZTA5XkEyXkFqcGdeQXVyNTAyODkwOQ@@._V1_SX300.jpg",
    "plot": "When a killer shark unleashes chaos on a beach community off Cape Cod, it's up to a local sheriff, a marine biologist, and an old seafarer to hunt the beast down.",
    "rated": "PG",
    "runtime": "124 min",
    "genre": "Adventure, Mystery, Thriller",
    "director": "Steven Spielberg",
    "writer": "Peter Benchley, Carl Gottlieb",
    "actors": "Roy Scheider, Robert Shaw, Richard Dreyfuss",
    "language": "English",
    "awards": "Won 3 Oscars. 15 wins & 20 nominations total",
    "ratings": [
      {
        "source": "Internet Movie Database",
        "value": "8.1/10",
        "score": 8.1
      },
      {
        "source": "Rotten Tomatoes",
        "value": "97%",
        "score": 97
      },
      {
        "source": "Metacritic",
        "value": "87/100",
        "score": 87
      }
    ],
    "imdbVotes": "644,091",
    "imdbID": "tt0073195",
    "boxOffice": "$265,859,065"
  },
  {
    "title": "The Silence of the Lambs",
    "year": 1991,
    "poster": "https://m.media-amazon.com/images/M/MV5BNjNhZTk0ZmEtNjJhMi00YzFlLWE1MmEtYzM1M2ZmMGMwMTU4XkEyXkFqcGdeQXVyNjU0OTQ0OTY@._V1_SX300.jpg",
    "plot": "A young F.B.I. cadet must receive the help of an incarcerated and manipulative cannibal killer to help catch another serial killer, a madman who skins his victims.",
    "rated": "R",
    "runtime": "118 min",
    "genre": "Crime, Drama, Thriller",
    "director": "Jonathan Demme",
    "writer": "Thomas Harris, Ted Tally",
    "actors": "Jodie Foster, Anthony Hopkins, Scott Glenn",
    "language": "English, Latin",
    "awards": "Won 5 Oscars. 70 wins & 50 nominations total",
    "ratings": [
      {
        "source": "Internet Movie Database",
        "value": "8.6/10",
        "score": 8.6
      },
      {
        "source": "Rotten Tomatoes",
        "value": "95%",
        "score": 95
      },
      {
        "source": "Metacritic",
        "value": "86/100",
        "score": 86
      }
    ],
    "imdbVotes": "1,505,680",
    "imdbID": "tt0102926",
    "boxOffice": "$130,742,922"
  },
  {
    "title": "Get Out",
    "year": 2017,
    "poster": "https://m.media-amazon.com/images/M/MV5BMjUxMDQwNjcyNl5BMl5BanBnXkFtZTgwNzcwMzc0MTI@._V1_SX300.jpg",
    "plot": "A young African-American visits his White girlfriend's parents for the weekend, where his simmering uneasiness about their reception of him eventually reaches a boiling point.",
    "rated": "R",
    "runtime": "104 min",
    "genre": "Horror, Mystery, Thriller",
    "director": "Jordan Peele",
    "writer": "Jordan Peele",
    "actors": "Daniel Kaluuya, Allison Williams, Bradley Whitford",
    "language": "English, Swahili",
    "awards": "Won 1 Oscar. 152 wins & 207 nominations total",
    "ratings": [
      {
        "source": "Internet Movie Database",
        "value": "7.8/10",
        "score": 7.8
      },
      {
        "source": "Rotten Tomatoes",
        "value": "98%",
        "score": 98
      },
      {
        "source": "Metacritic",
        "value": "85/100",
        "score": 85
      }
    ],
    "imdbVotes": "668,240",
    "imdbID": "tt5052448",
    "boxOffice": "$176,196,665"
  },
  {
    "title": "Psycho",
    "year": 1960,
    "poster": "https://m.media-amazon.com/images/M/MV5BNTQwNDM1YzItNDAxZC00NWY2LTk0M2UtNDIwNWI5OGUyNWUxXkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SX300.jpg",
    "plot": "A Phoenix secretary embezzles $40,000 from her employer's client, goes on the run and checks into a remote motel run by a young man under the domination of his mother.",
    "rated": "R",
    "runtime": "109 min",
    "genre": "Horror, Mystery, Thriller",
    "director": "Alfred Hitchcock",
    "writer": "Joseph Stefano, Robert Bloch",
    "actors": "Anthony Perkins, Janet Leigh, Vera Miles",
    "language": "English",
    "awards": "Nominated for 4 Oscars. 8 wins & 14 nominations total",
    "ratings": [
      {
        "source": "Internet Movie Database",
        "value": "8.5/10",
        "score": 8.5
      },
      {
        "source": "Rotten Tomatoes",
        "value": "97%",
        "score": 97
      },
      {
        "source": "Metacritic",
        "value": "97/100",
        "score": 97
      }
    ],
    "imdbVotes": "703,743",
    "imdbID": "tt0054215",
    "boxOffice": "$32,000,000"
  },
  {
    "title": "A Quiet Place",
    "year": 2018,
    "poster": "https://m.media-amazon.com/images/M/MV5BMjI0MDMzNTQ0M15BMl5BanBnXkFtZTgwMTM5NzM3NDM@._V1_SX300.jpg",
    "plot": "A family struggles for survival in a world where most humans have been killed by blind but noise-sensitive creatures. They are forced to communicate in sign language to keep the creatures at bay.",
    "rated": "PG-13",
    "runtime": "90 min",
    "genre": "Drama, Horror, Sci-Fi",
    "director": "John Krasinski",
    "writer": "Bryan Woods, Scott Beck, John Krasinski",
    "actors": "Emily Blunt, John Krasinski, Millicent Simmonds",
    "language": "American Sign , English",
    "awards": "Nominated for 1 Oscar. 37 wins & 128 nominations total",
    "ratings": [
      {
        "source": "Internet Movie Database",
        "value": "7.5/10",
        "score": 7.5
      },
      {
        "source": "Rotten Tomatoes",
        "value": "96%",
        "score": 96
      },
      {
        "source": "Metacritic",
        "value": "82/100",
        "score": 82
      }
    ],
    "imdbVotes": "565,622",
    "imdbID": "tt6644200",
    "boxOffice": "$188,024,361"
  },
  {
    "title": "Halloween",
    "year": 1978,
    "poster": "https://m.media-amazon.com/images/M/MV5BNzk1OGU2NmMtNTdhZC00NjdlLWE5YTMtZTQ0MGExZTQzOGQyXkEyXkFqcGdeQXVyMTQxNzMzNDI@._V1_SX300.jpg",
    "plot": "Fifteen years after murdering his sister on Halloween night 1963, Michael Myers escapes from a mental hospital and returns to the small town of Haddonfield, Illinois to kill again.",
    "rated": "R",
    "runtime": "91 min",
    "genre": "Horror, Thriller",
    "director": "John Carpenter",
    "writer": "John Carpenter, Debra Hill",
    "actors": "Donald Pleasence, Jamie Lee Curtis, Tony Moran",
    "language": "English",
    "awards": "9 wins & 3 nominations",
    "ratings": [
      {
        "source": "Internet Movie Database",
        "value": "7.7/10",
        "score": 7.7
      },
      {
        "source": "Rotten Tomatoes",
        "value": "96%",
        "score": 96
      },
      {
        "source": "Metacritic",
        "value": "87/100",
        "score": 87
      }
    ],
    "imdbVotes": "300,659",
    "imdbID": "tt0077651",
    "boxOffice": "$47,160,000"
  },
  {
    "title": "Shaun of the Dead",
    "year": 2004,
    "poster": "https://m.media-amazon.com/images/M/MV5BMTg5Mjk2NDMtZTk0Ny00YTQ0LWIzYWEtMWI5MGQ0Mjg1OTNkXkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SX300.jpg",
    "plot": "The uneventful, aimless lives of a London electronics salesman and his layabout roommate are disrupted by the zombie apocalypse.",
    "rated": "R",
    "runtime": "99 min",
    "genre": "Comedy, Horror",
    "director": "Edgar Wright",
    "writer": "Simon Pegg, Edgar Wright",
    "actors": "Simon Pegg, Nick Frost, Kate Ashfield",
    "language": "English",
    "awards": "Nominated for 3 BAFTA 14 wins & 20 nominations total",
    "ratings": [
      {
        "source": "Internet Movie Database",
        "value": "7.9/10",
        "score": 7.9
      },
      {
        "source": "Rotten Tomatoes",
        "value": "92%",
        "score": 92
      },
      {
        "source": "Metacritic",
        "value": "76/100",
        "score": 76
      }
    ],
    "imdbVotes": "582,569",
    "imdbID": "tt0365748",
    "boxOffice": "$13,542,874"
  }
]
</pre>
`;
